packages/snaps-controllers/src/utils.ts

```diff
@@ -38,6 +38,28 @@ export function setDiff<
   ) as Diff<ObjectA, ObjectB>;
 }
 
+/**
+ * Takes two objects and does a Set Intersection of them.
+ *
+ * @param objectA - The object on which the difference is being calculated.
+ * @param objectB - The object whose properties will be removed from objectA.
+ * @returns The elements that are in both sets (ObjectA and ObjectB).
+ */
+export function setIntersection<
+  ObjectA extends Record<any, unknown>,
+  ObjectB extends Record<any, unknown>,
+>(objectA: ObjectA, objectB: ObjectB): Intersection<ObjectA, ObjectB> {
+  return Object.entries(objectA).reduce<Record<any, unknown>>(
+    (acc, [key, value]) => {
```

Comment:
Please test this function individually too

Reply:
Added.